src/shared/osclib/__init__.py

```diff
@@ -152,115 +152,44 @@ def are_on_same_machine(url1: str, url2: str) -> bool:
-                    == socket.gethostbyaddr(hostname_2)):
-                return True
+    def resolve_host(name):
+        for family in (socket.AF_INET, socket.AF_INET6):
+            result = None
+            try:
+                result = socket.getaddrinfo(name, None, family, socket.SOCK_DGRAM)
+            except socket.gaierror:
+                continue
+            if result:
+                return result[0][4][0]
+        return host
 
-        except BaseException as e:
-            _logger.warning(
-                f'Failed to find IPv6 host by addr '
-                f'for "{hostname_1}" or "{hostname_2}"\n'
-                f'{str(e)}')
+    host1 = resolve_host(address1.hostname)
+    host2 = resolve_host(address2.hostname)
 
-    elif not addr1_ipv6 and not addr2_ipv6:
-        try:
-            addr1 = socket.gethostbyname(hostname_1)
-            addr2 = socket.gethostbyname(hostname_2)
-            if (addr1 in ('127.0.0.1', '127.0.1.1')
-                    and addr2 in ('127.0.0.1', '127.0.1.1')):
-                return True
+    if host1 == host2:
+        return True
 
-            try:
-                if socket.gethostbyaddr(addr1) == socket.gethostbyaddr(addr2):
-                    return True
-
-            except BaseException as e:
-                _logger.warning(
-                    f'Failed to find host by addr'
-                    f'for "{addr1}" or "{addr2}"\n'
-                    f'{str(e)}')
-
-        except BaseException as e:
-            _logger.warning(
-                f'Failed to find host by name for '
-                f'"{hostname_1}" or "{hostname_2}"\n'
-                f'{str(e)}')
-            
-    # if (socket.gethostbyname(address1.hostname)
-    #             in ('127.0.0.1', '127.0.1.1')
-    #         and socket.gethostbyname(address2.hostname)
-    #             in ('127.0.0.1', '127.0.1.1')):
-    #     return True            
-    
-    elif addr1_ipv6:
-        try:
-            if (socket.gethostbyaddr(hostname_1)[0] == 'localhost'
-                    and socket.gethostbyname(hostname_2)
-                        in ('127.0.0.1', '127.0.1.1')):
-                return True
-
-        except BaseException as e:
-            _logger.warning(str(e))
-            
-    elif addr2_ipv6:
-        try:
-            if (socket.gethostbyaddr(hostname_2)[0] == 'localhost'
-                    and socket.gethostbyname(hostname_1)
-                        in ('127.0.0.1', '127.0.1.1')):
-                return True
+    LOCAL_ADDRS = ('127.0.0.1', '127.0.1.1', '::1')
 
-        except BaseException as e:
-            _logger.warning(str(e))
+    if host1 in LOCAL_ADDRS and host2 in LOCAL_ADDRS:
+        return True
 
     ip = get_machine_192()
 
-    print('are_on_same_machine', url1, url2)
-    print(f"    '{ip}' '{address1.hostname}' '{address2.hostname}")
-
-    if ip not in (hostname_1, hostname_2):
+    if ip not in (resolve_host(address1.hostname), resolve_host(address2.hostname)):
```

Comment:
Very personal preference, I prefer when lines does not exceed 79 characters. Not strong of course.

Reply:
updated, splitted on two lines